node/src/BaseClient.ts

```diff
@@ -1079,8 +1086,8 @@ export class BaseClient {
      * ```
      */
     public async set(
-        key: string | Uint8Array,
-        value: string | Uint8Array,
+        key: GlideString | Uint8Array,
+        value: GlideString | Uint8Array,
         options?: SetOptions,
     ): Promise<"OK" | string | null> {
         return this.createWritePromise(createSet(key, value, options));
```

Comment:
We need to send a Decoder here, since `set` can return the previous result (which can be binary or string).